python/ray/tests/conftest.py

```diff
@@ -287,6 +303,24 @@ def log_pubsub():
     p.close()
 
 
+@contextmanager
+def manage_tls(use_tls):
+    if use_tls:
+        key_filepath, cert_filepath, temp_dir = setup_tls()
+    yield use_tls
```

Comment:
Should here be a try-finally, so that `teardown_tls` is called also when `yield` experiences an exceptions? I know this is just tests, but still.